### PR TITLE
Configuration for opening bitstream formats as attachment

### PR DESCRIPTION
**Pull request by nona-luypaert**
## Description
This PR makes it possible to configure which files should always be downloaded, based on file format. Adding the mimetype/extension of a format to the configuration will ensure the browser will NOT open these files inline.

#### Changes
- Added new property to dspace.cfg: `webui.content_disposition_format`. By default this only contains the RTF-format.
- Edited `retrieve` method in BitstreamRestController to set `content-disposition: attachment` in the Response Headers if the file's mimetype or extension appears in the config property.
- Added extra test to BitstreamRestControllerIT, along with some extra configured formats in local.cfg.

#### Intructions
- First, build DSpace without any values in `webui.content_disposition_format`.
- Create an Item and add some files to it, e.g. a PDF- and RTF-file.
- Try to download them and verify the files are opened inline.
- Build DSpace again, this time with `webui.content_disposition_format = text/richtext, pdf`
(`rtf, pdf` ; `rtf, application/pdf` ; `text/richtext, application/pdf` would also work).
- Try to download the same files, and verify they get indeed downloaded.

## Checklist
- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new libraries/dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies REST API endpoints, I've opened a separate [REST Contract](https://github.com/DSpace/RestContract/blob/main/README.md) PR related to this change.
- [x] If my PR includes new configurations, I've provided basic technical documentation in the PR itself.
- [x] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

**Review by abollini — CHANGES_REQUESTED** [review]
thanks for this contribution @nona-luypaert it looks to be very useful and I recommend to try at our best to have it in 7.6

I spot a NPE in an edge scenario that should be fixed (it would be also nice to have a specific test for that... but if you are short of time it is ok to skip for me)

**Comment by alanorth** [comment]
Currently has merge conflicts against `main`.

Regarding the proposed functionality, it looks like this might fix a related bug with content disposition. In previous DSpace releases, bitstreams would _always_ be opened inline _unless they were larger than `webui.content_disposition_threshold`_. In 7.6-snapshot this is true right now for bitstreams like PDFs, but I've noticed that image bitstreams _always_ download due to a null `Content-Disposition` header, for example a PDF thumbnail that I tried to open in a new tab in Firefox:

```
content-disposition
	null;filename="=?UTF-8?Q?AfricaRISING=5FWA=5FTechnical_Report_USAID-IITA_(?= =?UTF-8?Q?October_2021_=E2=80=93_31_March_2022)=5FPJ-1503.pdf.jpg?="
```

I will test once @nona-luypaert rebases the patch.

**Comment by alanorth** [comment]
Reviewed by testing before and after adding `application/pdf` to the new `webui.content_disposition_format` setting:

- Before: PDF opens inline
- After: PDF is downloaded

So this works as advertised. :+1:

I would suggest adding a few more words to the comment for the new `webui.content_disposition_format` setting in dspace.cfg, perhaps:

<blockquote><pre># Set which mimetypes, file extensions will NOT be opened inline.
# Files with these mimetypes/extensions will always be downloaded
# regardless of the threshold above.
webui.content_disposition_format = text/richtext</pre></blockquote>

<hr />
Also note, this *does not* fix the issue with images being forced to download when doing right click → open in new tab. That is unrelated to this PR, but I had hoped it might fix it as a bonus. I will open a new issue for that.

File: dspace/config/dspace.cfg [hunk 1]
```diff
@@ -1353,6 +1353,11 @@ websvc.opensearch.formats = html,atom,rss
 # Use -1 to force all bitstream to be served inline
 webui.content_disposition_threshold = 8388608
 
+#### Content Attachment Disposition Formats ####
+#
+# Set which mimetypes, file extensions will NOT be opened inline
+# Files with these mimetypes/extensions will always be downloaded
```
Comment (alanorth) [inline dspace/config/dspace.cfg:1359]: I would suggest continuing this comment to say that files with these mimetypes/extensions will always be downloaded regardless of the content disposition threshold.

**Review by alanorth — APPROVED** [review]

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/BitstreamRestControllerIT.java [hunk 2]
```diff
@@ -1235,4 +1236,51 @@ public void closeInputStreamsDownloadWithCoverPage() throws Exception {
         Mockito.verify(inputStreamSpy, times(1)).close();
     }
 
+
+    @Test
+    public void checkContentDispositionOfFormats() throws Exception {
+        context.turnOffAuthorisationSystem();
```
Comment (tdonohue) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/BitstreamRestControllerIT.java:1242]: As mentioned above, I'd recommend moving the custom configuration for `webui.content_disposition_format = text/richtext, text/xml, txt` to this test method itself.  

You should be able to easily do that as follows:
```
configurationService.setProperty("webui.content_disposition_format", new String[] {
            "text/richtext",
            "text/xml",
            "txt"
        });
```

Configurations are reloaded after every test runs, so this should be all you need... the `webui.content_disposition_format` setting will go back to its default value for the next test.

**Review by tdonohue — COMMENTED** [review]
@nona-luypaert : Overall this looks great & it works well.  I tested a small PDF beforehand and saw it opened inline. But, when setting the new config to simply `webui.content_disposition_format = pdf` the same PDF downloaded.   (As a bonus, I noticed that you don't even need to restart Tomcat after changing this config... the new settings reload immediately)

My only comments are on the test configuration.  I think it's cleaner to move your test configs to the method that needs them.  That way they are not used by every other test.  See inline comments below for how to do that.   Overall, I'm +1, but I'd appreciate it if the test configs could be moved.

**Review by tdonohue — APPROVED** [review]
👍 Thanks @nona-luypaert !  This looks good to me now & works well.

**Pull request merged**